packages/fxa-auth-server/lib/routes/utils/oauth.js

```diff
@@ -48,6 +47,13 @@ module.exports = {
         .toString('hex');
     }
 
+    if (!credentials.id && grant.session_token) {
+      // if the caller also requested apps/oldsync scope and a refresh token, then
+      // we will end up creating a placeholder device record for that refresh token.
+      // We should ensure that the sessionToken also gets attached to that same device record.
+      credentials.id = grant.session_token;
```

Comment:
Ugh, this code is correct, but it's also super non-obvious that the `id` field of this object should contain the sessionToken id. I guess this is a side-effect of passing around raw Hapi credential objects in order places. Maybe worth a comment for uninitiated readers?